### DISCUSSION TITLE
Creating & Renaming Workspaces

### DISCUSSION THREAD

**Discussion by munkyjunky** [structurizr/onpremises#68] [Q&A] [answered]
I'm trying to get a CI/CD set up with on premises Structurizr. So far I've got a system that will go through and publish all the workspace.dsl files into the corresponding workspaces, but this relies on me manually creating a workspace and configuring it beforehand. Ideally, I'd like to automate as much of this as possible but I can't seem to find any documentation for the APIs on how to do this. 

For creating a workspace, I can call `POST /api/workspace` to create a new workspace, but this names it as `Workspace 0002` or similar, with an empty description. I also can't find any UI page to log in and change the workspace once it's been created. It appears the only solution is manually editing the `workspace.json` and `workspace.properties` files in the S3 storage to set these values, which is really not ideal at all. 

I've seen #61, which to me would suggest there's no UI for this at the moment because there's no API to handle this. Am I missing something obvious for getting these workspaces renamed? Or is there a way to programatically create a workspace with a name & description?

**Comment by simonbrowndotje** [comment, marked as answer]
Once you've created the workspace and have the API key/secret, you can use that to just push a workspace definition (e.g. via the CLI or your own custom tooling). If you're using the DSL, you can set the workspace name via the `workspace` keyword -> https://docs.structurizr.com/dsl/language#workspace

**Reply by munkyjunky** [reply]
Great, I thought I must be missing something obvious! Thank you.
